packages/flutter/lib/src/animation/animation_controller.dart

```diff
@@ -38,6 +38,10 @@ const Tolerance _kFlingTolerance = const Tolerance(
   distance: 0.01,
 );
 
+/// Don't start the animation if the target value is within this tolerance of
+/// the current value.
+const Tolerance _kAnimateToTolerance = const Tolerance();
```

Comment:
That makes me wonder if we should remove the tolerance from the animation controller altogether  and check in the ScrollPosition's animateTo if we are asked to animate to roughly the same position we're already at. And if that's the case, just redirect to the ScrollPosition's jumpTo.